packages/multichain/src/scope/supported.ts

```diff
@@ -14,99 +18,141 @@ import { parseScopeString } from './types';
-  getInternalAccounts: () => { type: string; address: string }[],
+  {
+    getEvmInternalAccounts,
+    getNonEvmAccountAddresses,
+  }: {
+    getEvmInternalAccounts: () => { type: string; address: Hex }[];
+    getNonEvmAccountAddresses: (scope: CaipChainId) => string[];
+  },
 ) => {
   const {
     address,
+    chainId,
     chain: { namespace, reference },
   } = parseCaipAccountId(account);
 
   const isSupportedEip155Account = () =>
-    getInternalAccounts().some(
+    getEvmInternalAccounts().some(
       (internalAccount) =>
         ['eip155:eoa', 'eip155:erc4337'].includes(internalAccount.type) &&
         isEqualCaseInsensitive(address, internalAccount.address),
     );
 
+  const isSupportedNonEvmAccount = () =>
+    getNonEvmAccountAddresses(chainId).includes(account);
+
   switch (namespace) {
     case KnownCaipNamespace.Wallet:
-      return reference === KnownCaipNamespace.Eip155
-        ? isSupportedEip155Account()
-        : false;
+      if (reference === KnownCaipNamespace.Eip155) {
+        return isSupportedEip155Account();
+      }
+      return isSupportedNonEvmAccount();
     case KnownCaipNamespace.Eip155:
       return isSupportedEip155Account();
     default:
-      return false;
+      return isSupportedNonEvmAccount();
   }
 };
 
 /**
  * Determines if a method is supported by the wallet.
+ *
  * @param scopeString - The scope string to check.
  * @param method - The method to check.
+ * @param hooks - An object containing the following properties:
+ * @param hooks.getNonEvmSupportedMethods - A function that returns the supported methods for a non EVM scope.
  * @returns A boolean indicating if the method is supported by the wallet.
  */
 export const isSupportedMethod = (
   scopeString: ExternalScopeString,
   method: string,
+  {
+    getNonEvmSupportedMethods,
+  }: {
+    getNonEvmSupportedMethods: (scope: CaipChainId) => string[];
+  },
 ): boolean => {
   const { namespace, reference } = parseScopeString(scopeString);
 
-  if (!namespace || !isKnownCaipNamespace(namespace)) {
+  if (!namespace) {
     return false;
   }
 
+  const isSupportedNonEvmMethod = () =>
+    isCaipChainId(scopeString) &&
+    getNonEvmSupportedMethods(scopeString).includes(method);
+
   if (namespace === KnownCaipNamespace.Wallet) {
     if (reference) {
-      if (
-        !isKnownCaipNamespace(reference) ||
-        reference === KnownCaipNamespace.Wallet
-      ) {
-        return false;
+      if (reference === KnownCaipNamespace.Eip155) {
+        return KnownWalletNamespaceRpcMethods[reference].includes(method);
       }
-      return KnownWalletNamespaceRpcMethods[reference].includes(method);
+      return isSupportedNonEvmMethod();
     }
 
     return KnownWalletRpcMethods.includes(method);
+  } else if (namespace === KnownCaipNamespace.Eip155) {
+    return KnownRpcMethods[namespace].includes(method);
   }
-
-  return KnownRpcMethods[namespace].includes(method);
+  return isSupportedNonEvmMethod();
```

Comment:
what are your thoughts of refactoring the following statements as such

```suggestion
if (namespace === KnownCaipNamespace.Wallet) {
  if (!reference) {
    return KnownWalletRpcMethods.includes(method);
  }

  if (reference === KnownCaipNamespace.Eip155) {
    return KnownWalletNamespaceRpcMethods[reference].includes(method);
  }

  return isSupportedNonEvmMethod();
}

if (namespace === KnownCaipNamespace.Eip155) {
  return KnownRpcMethods[namespace].includes(method);
}

return isSupportedNonEvmMethod();
```

taking advantage of early returns to reduce the cognitive load of reading through more nested if conditions ?

Reply:
Makes sense! Updated in [e1eb06a](https://github.com/MetaMask/core/pull/5191/commits/e1eb06a976c5b980251dc7190b65af75a51911d9).